File: app/src/main/java/com/task/paginationlist/presentation/piclist/views/adapters/PicFragmentPagerAdapter.java
package com.task.paginationlist.presentation.piclist.views.adapters;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.util.Pair;

import com.task.paginationlist.presentation.piclist.views.fragments.PictureFragment;
import com.task.paginationlist.presentation.piclist.views.paginator.Config;

public class PicFragmentPagerAdapter extends FragmentStatePagerAdapter {
    private int count;

    public PicFragmentPagerAdapter(FragmentManager fm, int count) {
        super(fm);
        this.count = count;
    }

    @Override
    public Fragment getItem(int position) {
        return PictureFragment.createPictureFragment(getPageAndPosionOfDbElement(position));
    }

    public void setCount(int count) {
        this.count = count;
        notifyDataSetChanged();
    }

    private Pair<Integer, Integer> getPageAndPosionOfDbElement(int pos) {
        int page = pos / Config.LIMIT + 1;
        int dbPosition = pos % Config.LIMIT;
        return new Pair<>(page, dbPosition);
    }

    @Override
    public int getCount() {
        return count;
    }
}

File: app/src/main/java/com/task/paginationlist/presentation/piclist/views/adapters/PicFragmentPagerAdapter.kt
package com.task.paginationlist.presentation.piclist.views.adapters

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import android.support.v4.util.Pair

import com.task.paginationlist.presentation.piclist.views.fragments.PictureFragment
import com.task.paginationlist.presentation.piclist.views.paginator.Config

class PicFragmentPagerAdapter(fm: FragmentManager, private var count: Int) : FragmentStatePagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return PictureFragment.createPictureFragment(getPageAndPosionOfDbElement(position))
    }

    fun setCount(count: Int) {
        this.count = count
        notifyDataSetChanged()
    }

    private fun getPageAndPosionOfDbElement(pos: Int): Pair<Int, Int> {
        val page = pos / Config.LIMIT + 1
        val dbPosition = pos % Config.LIMIT
        return Pair(page, dbPosition)
    }

    override fun getCount(): Int {
        return count
    }
}
